langchain4j/src/main/java/dev/langchain4j/service/tool/ToolService.java

```diff
@@ -97,9 +97,25 @@ public void tools(Collection<Object> objectsWithTools) {
                 throw illegalConfiguration("Tool '%s' must be an object, not a class", objectWithTool);
             }
 
+            if (objectWithTool instanceof Iterable) {
+                throw illegalConfiguration(
+                        "Tool '%s' is an Iterable (likely a nested collection). "
+                                + "Please pass tool objects directly, not wrapped in collections.",
+                        objectWithTool.getClass().getName());
+            }
+
+            boolean hasToolMethod = false;
             for (Method method : objectWithTool.getClass().getDeclaredMethods()) {
-                getAnnotatedMethod(method, Tool.class)
-                        .ifPresent(toolMethod -> processToolMethod(objectWithTool, toolMethod));
+                if (getAnnotatedMethod(method, Tool.class).isPresent()) {
+                    hasToolMethod = true;
+                    processToolMethod(objectWithTool, method);
```

Comment:
```suggestion
                    processToolMethod(objectWithTool, toolMethod);
```

There are currently 2 tests failing because of this: `AiServicesWithToolsIT.tools_as_agents_tests`

Reply:
I'll check this. Thanks. Locally it was passing.

Reply:
I will fix it after the merge